### PR TITLE
update MsmsSweeps tests in verification.py

### PR DESCRIPTION
**Pull request by gbinux**
Fixed some bugs in MsmsSweeps test class and added a few tests.

The tests used to generate the plots in #1173 are:

1. `test_selective_discoal_vs_msp_use_discoal_paper_param`
2. `test_selective_msms_vs_discoal_use_discoal_paper_param`
3. `test_selective_msms_vs_msp_use_discoal_paper_param`

### CODE REVIEW

File: data/Makefile [hunk 1]
```diff
@@ -36,6 +36,10 @@ msHOT:
 discoal:
 	wget https://github.com/kern-lab/discoal/archive/v0.1.5.tar.gz
 	tar -xf v0.1.5.tar.gz
+	# To allow larger num_sites, a macro in the header file is updated
+	sed -e 's/^#define MAXSITES.*$$/#define MAXSITES 500000/' \
```
Comment (jeromekelleher) [inline data/Makefile:40]: Could use ``sed -i`` here to do it in-place?

File: verification.py [hunk 2]
```diff
@@ -845,6 +845,14 @@ class MsmsSweeps(Test):
     use h=0.5 to calculate that for Aa.
     """
 
+    def _set_cmp_to_discoal(self, cmp_to_discoal):
```
Comment (jeromekelleher) [inline verification.py:848]: I'm not crazy about this pattern, modifying the state of the Test objects isn't a great idea for various reasons. Is there some reason we can't pass ``cmp_to_discoal`` to the calling function?

**Review by jeromekelleher — COMMENTED** [review]
Looks good, thanks @gbinux, minor comments.

**Comment by gbinux** [comment]
`test_selective_..._use_discoal_paper_param` tests match with parameters used in the `discoal` paper, but use large num_sites and are quite slow. Please let me know if I should hide them away or keep them.

**Comment by jeromekelleher** [comment]
> test_selective_..._use_discoal_paper_param tests match with parameters used in the discoal paper, but use large num_sites and are quite slow. Please let me know if I should hide them away or keep them.

How slow? If it takes more than a minute, maybe we could reduce the number of sites a bit (and maybe put in a comment saying "set sites to X" to reproduce results from paper?)

**Comment by gbinux** [comment]
> How slow? If it takes more than a minute, maybe we could reduce the number of sites a bit (and maybe put in a comment saying "set sites to X" to reproduce results from paper?)

longer than 1 minute. I will update them as you recommended above @jeromekelleher

**Comment by jeromekelleher** [comment]
(ps. this PR is still marked as a draft so I'm assuming you don't want it merged yet @gbinux)

**Comment by gbinux** [comment]
Commits were squashed into a single one (added tests into the MsmsSweeps test class and fixed a few bugs/typos.) @jeromekelleher

**Review by jeromekelleher — APPROVED** [review]
Great, thanks @gbinux !

**Pull request merged**